### PR TITLE
Validation

### PR DESCRIPTION
**Pull request by jonnorstrom**
Ready for review.

### CODE REVIEW

**Comment by jonnorstrom** [comment]
Coveralls is saying it failed because previous to this, it thought we had 100% coverage, which we did not. Safe to marge in terms of tests.

File: app/assets/stylesheets/application.css [hunk 1]
```diff
@@ -23,24 +23,9 @@
    color: red;
  }
 
+/* css for error exclamation point */
  .excl {
   background-color: black;
   padding: 6px;
   border-radius: 100%;
  }
-
```
Comment (DocOnDev) [inline app/assets/stylesheets/application.css]: What is the thinking behind breaking these out into separate files? I am not opposed to it, just curious about your reasoning.

File: spec/models/user_spec.rb [hunk 2]
```diff
@@ -8,9 +8,25 @@
     it { should validate_presence_of(col) }
   end
 
+  it { should validate_length_of(:username).is_at_least(MIN_USERNAME_LENGTH)}
+  it { should validate_length_of(:username).is_at_most(MAX_USERNAME_LENGTH)}
+  it { should have_secure_password }
+  VALID_USERNAME_ARRAY.each do |username|
```
Comment (DocOnDev) [inline spec/models/user_spec.rb]: "username" seems a misleading variable name here. What are each of these VALID_USERNAME_ARRAY items? Not a username, but valid_characters (or what)?

File: app/assets/stylesheets/application.css (hunk 1 above)
Comment (jonnorstrom) [reply app/assets/stylesheets/application.css]: Once applications get large - having your CSS structured similarly to your views is helpful. This is just some preemptive organization.

File: spec/models/user_spec.rb (hunk 2 above)
Comment (jonnorstrom) [reply spec/models/user_spec.rb]: Nope, they are different usernames that all happen to be valid (e.g. No special characters, varying lengths all within 6-20). We also made an INVALID array.

File: app/assets/stylesheets/application.css (hunk 1 above)
Comment (DocOnDev) [reply app/assets/stylesheets/application.css]: Cool. I support this. Just checking in with you two on reasoning.

**Comment by jonnorstrom** [comment]
@DocOnDev merged some refactoring into this branch. No forward progress was made, but changed the structure of some of our tests, etc. You may want to review again before merging. @SeanMNorton just went through it all with me and supports it.

**Pull request merged**